jobs/transforms/exp_model_audits_test.py

```diff
@@ -0,0 +1,249 @@
+                'The following extra attributes are present: invalid')
+        ])
+
+    def test_validate_exp_model_object_with_invalid_exploration_property(self):
+        invalid_commit_cmd_model = exp_models.ExplorationSnapshotMetadataModel(
+            id='model_id-1',
+            created_on=self.YEAR_AGO,
+            last_updated=self.NOW,
+            committer_id='committer_id',
+            commit_type='create',
+            commit_cmds_user_ids=[
+                'commit_cmds_user_1_id', 'commit_cmds_user_2_id'],
+            content_user_ids=['content_user_1_id', 'content_user_2_id'],
+            commit_cmds=[{
+                'cmd': 'edit_exploration_property',
+                'property_name': 'invalid',
+                'old_value': 'old_value',
+                'new_value': 'new_value',
+            }])
+
+        output = (
+            self.pipeline
+            | beam.Create([invalid_commit_cmd_model])
+            | beam.ParDo(
+                exp_model_audits
+                .ValidateExplorationCommitCmdsSchema())
+        )
+
+        self.assert_pcoll_equal(output, [
+            audit_errors.CommitCmdsValidateError(
+                invalid_commit_cmd_model,
+                '{u\'new_value\': u\'new_value\', u\'cmd\': '
+                'u\'edit_exploration_property\', u\'old_value\': '
+                'u\'old_value\', u\'property_name\': u\'invalid\'}',
+                'Value for property_name in cmd edit_exploration_property: '
+                'invalid is not allowed')
+        ])
+
+    def test_validate_exp_model_object_with_invalid_state_property(self):
+        invalid_commit_cmd_model = exp_models.ExplorationSnapshotMetadataModel(
+            id='model_id-1',
+            created_on=self.YEAR_AGO,
+            last_updated=self.NOW,
+            committer_id='committer_id',
+            commit_type='create',
+            commit_cmds_user_ids=[
+                'commit_cmds_user_1_id', 'commit_cmds_user_2_id'],
+            content_user_ids=['content_user_1_id', 'content_user_2_id'],
+            commit_cmds=[{
+                'cmd': 'edit_state_property',
+                'state_name': 'state_name',
+                'property_name': 'invalid',
+                'old_value': 'old_value',
+                'new_value': 'new_value',
+            }])
+
+        output = (
+            self.pipeline
+            | beam.Create([invalid_commit_cmd_model])
+            | beam.ParDo(
+                exp_model_audits
+                .ValidateExplorationCommitCmdsSchema())
+        )
+
+        self.assert_pcoll_equal(output, [
+            audit_errors.CommitCmdsValidateError(
+                invalid_commit_cmd_model,
+                '{u\'new_value\': u\'new_value\', u\'cmd\': '
+                'u\'edit_state_property\', u\'old_value\': u\'old_value\', '
+                'u\'state_name\': u\'state_name\', '
+                'u\'property_name\': u\'invalid\'}',
```

Comment:
Can we work directly with a dict instead of working with string containing the dict? This will probably break in Python 3 since the ordering of items in dicts is not stable in Python 3.

Reply:
Done